components/scream/src/share/field/field_header.hpp

```diff
@@ -98,16 +98,11 @@ class FieldHeader : public ekat::enable_shared_from_this<FieldHeader> {
   // Allocation properties
   FieldAllocProp                  m_alloc_prop;
 
-  // If this field is a sub-view of another field, we keep a pointer to the parent
-  // OTOH, if other fields are sub-view of this field, we keep a pointer to them
-  std::weak_ptr<FieldHeader>              m_parent;
-  std::list<std::weak_ptr<FieldHeader>>   m_children;
-
   // Extra data associated with this field
   extra_data_type                 m_extra_data;
 };
 
-// Use this free function to exploit features of enable_from_this
+// Use this free functions to exploit features of enable_from_this
```

Comment:
Does this comment apply to the two functions below?

Reply:
Ah, well, yes and no. You do need those two fcns to trigger `enable_shared_from_this` extra capabilities. But the second should be the only way to create a subfield header...except that I left the subfield header ctor declared (but never defined!).

I will revert back to singular, and make a separate comment for the 2nd fcn.